alma/alma-8.x/common/install_nvidiagpudriver.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+set -ex
+
+# Install Cuda
+NVIDIA_VERSION="510.85.02"
+CUDA_VERSION="11-6"
+dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel8/x86_64/cuda-rhel8.repo
+dnf clean expire-cache
+dnf install cuda-toolkit-11-6 -y
+echo 'export PATH=$PATH:/usr/local/cuda/bin' | tee -a /etc/bash.bashrc
+echo 'export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64' | tee -a /etc/bash.bashrc
+$COMMON_DIR/write_component_version.sh "CUDA" ${CUDA_VERSION}
+
+# Download CUDA samples
+CUDA_SAMPLES_VERSION="11.6"
+TARBALL="v${CUDA_SAMPLES_VERSION}.tar.gz"
+CUDA_SAMPLES_DOWNLOAD_URL=https://github.com/NVIDIA/cuda-samples/archive/refs/tags/${TARBALL}
+wget ${CUDA_SAMPLES_DOWNLOAD_URL}
+tar -xvf ${TARBALL}
+pushd ./cuda-samples-${CUDA_SAMPLES_VERSION}
+make -j $(nproc)
```

Comment:
Do we want to specify specific hardware targets to compile for? It looks like we can specify architectures like: 
make SMS="50 60"

https://github.com/NVIDIA/cuda-samples/blob/master/README.md

Reply:
Yes, we could. But is there a reason? None of the other compilations are targeted to a specific architecture.
@jithinjosepkl @jeseszhang1010

Reply:
The main benefit is that it would speed up image creation time, but perhaps it doesn't matter.